src/test/app/Escrow_test.cpp

```diff
@@ -671,18 +674,84 @@ struct Escrow_test : public beast::unit_test::suite
+            env.fund(XRP(5000), alice, carol);
+            auto const seq = env.seq(alice);
+
+            env(condpay(alice, carol, XRP(1000), makeSlice(cb1), env.now() + 1s));
+
+            auto const m = env.meta();
+            BEAST_EXPECT((*m)[sfTransactionResult] == tesSUCCESS);
+
+            auto const escrow = env.le(keylet::escrow(alice.id(), seq));
+            BEAST_EXPECT(escrow);
+
+            // Verify that Alice's owner directory tracks the escrow entry
+            ripple::Dir aod (*env.current(), keylet::ownerDir(alice.id()));
+            BEAST_EXPECT(std::distance(aod.begin(), aod.end()) == 1);
+            BEAST_EXPECT(std::find(aod.begin(), aod.end(), escrow) != aod.end());
+
+            // Verify that Carol doesn't have an owner directory at all.
+            ripple::Dir cod (*env.current(), keylet::ownerDir(carol.id()));
+            BEAST_EXPECT(cod.begin() == cod.end());
+        }
+
+        {
+            testcase ("Metadata (with fix1523, to self)");
+
+            Env env(*this, with_features(featureEscrow));
+            env.fund(XRP(5000), alice, carol);
+            auto const seq = env.seq(alice);
+
+            env(condpay(alice, alice, XRP(1000), makeSlice(cb1), env.now() + 1s));
+
+            auto const m = env.meta();
+            BEAST_EXPECT((*m)[sfTransactionResult] == tesSUCCESS);
+
+            auto const escrow = env.le(keylet::escrow(alice.id(), seq));
+            BEAST_EXPECT(escrow);
+
+            // Verify that Alice's owner directory tracks the escrow entry
+            ripple::Dir aod (*env.current(), keylet::ownerDir(alice.id()));
+            BEAST_EXPECT(std::distance(aod.begin(), aod.end()) == 1);
+            BEAST_EXPECT(std::find(aod.begin(), aod.end(), escrow) != aod.end());
+        }
+
+        {
+            testcase ("Metadata (with fix1523, to other)");
+
+            Env env(*this, with_features(featureEscrow, fix1523));
+            env.fund(XRP(5000), alice, carol);
+            auto const seq = env.seq(alice);
+
+            env(condpay(alice, carol, XRP(1000), makeSlice(cb1), env.now() + 1s));
+
+            auto const m = env.meta();
+            BEAST_EXPECT((*m)[sfTransactionResult] == tesSUCCESS);
+
+            auto const escrow = env.le(keylet::escrow(alice.id(), seq));
+            BEAST_EXPECT(escrow);
+
+            // Verify that Alice's owner directory tracks the escrow entry
+            ripple::Dir aod (*env.current(), keylet::ownerDir(alice.id()));
+            BEAST_EXPECT(std::distance(aod.begin(), aod.end()) == 1);
+            BEAST_EXPECT(std::find(aod.begin(), aod.end(), escrow) != aod.end());
+
+            // Verify that Carol's owner directory tracks the escrow entry too.
+            ripple::Dir cod (*env.current(), keylet::ownerDir(carol.id()));
+            BEAST_EXPECT(std::distance(cod.begin(), cod.end()) == 1);
+            BEAST_EXPECT(std::find(cod.begin(), cod.end(), escrow) != cod.end());
```

Comment:
Is it worth adding a test to confirm `EscrowFinish` and `EscrowCancel` properly delete the entries in both directories?

Reply:
:+1: